### PR TITLE
perf: pre-allocate buffer to avoid per-call zeroing of MLA KV counter buffer

### PR DESCRIPTION
**Pull request by jdebache**
## Purpose

Reuses a caller-owned FlashInfer MLA multi-CTA KV counter buffer, eliminating a fresh `torch.zeros()` allocation on every decode step while growing the buffer on demand.

This is allowed per the documentation of `trtllm_batch_decode_with_kv_cache_mla` in FlashInfer:
> ... and be zero-initialized **once**. ...

## Test Plan

Full GSM8K A/B testing with DeepSeek-V2-Lite-Chat.

## Test Result

Showed no measurable accuracy regression across two GB300 runs:

| Same-GPU comparison | Baseline | Patched | Delta |
|---|---:|---:|---:|
| GB300 GPU 0 | 833/1,319 — 63.1539% | 834/1,319 — 63.2297% | +0.0758 pp |
| GB300 GPU 1 | 846/1,319 — 64.1395% | 844/1,319 — 63.9879% | −0.1516 pp |
| Mean | 63.6467% | 63.6088% | **−0.0379 pp** |

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [X] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [X] The test plan, such as providing test command.
- [X] The test results, such as pasting the results comparison before and after, or e2e results
- [NA] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: vllm/v1/attention/backends/mla/flashinfer_mla.py [hunk 1]
```diff
@@ -48,6 +52,24 @@ def _get_workspace_buffer(return_lse: bool) -> torch.Tensor:
     return _fi_workspace
 
 
+_fi_kv_counter: torch.Tensor | None = None
+_fi_sm_count: int | None = None
+
+
+def _get_kv_counter_buffer(
+    batch_size: int, num_qo_heads: int, device: torch.device
+) -> torch.Tensor:
```
Comment (pavanimajety) [inline vllm/v1/attention/backends/mla/flashinfer_mla.py:61]: nit: Could you please add a doc string? What this buffer is being used for?

**Review by pavanimajety — APPROVED** [review]
LGTM, thanks for your contribution!

File: vllm/v1/attention/backends/mla/flashinfer_mla.py (hunk 1 above)
Comment (pavanimajety) [inline vllm/v1/attention/backends/mla/flashinfer_mla.py:61]:
```suggestion
) -> torch.Tensor:
    """Return the MLA KV counter buffer.
    
    ``kv_counter_buffer`` is used by the
    ``trtllm_batch_decode_with_kv_cache_mla`` kernel to initialize
    ``multi_ctas_kv_counter_buffer`` in the first iteration. It only needs to be
    zero-initialized for the first iteration; later iterations zero it in the
    kernel epilogue.
    """
```

File: vllm/v1/attention/backends/mla/flashinfer_mla.py [hunk 2]
```diff
@@ -48,6 +52,32 @@ def _get_workspace_buffer(return_lse: bool) -> torch.Tensor:
...
+    """Return the MLA KV counter buffer.
+
```
Comment (pavanimajety) [inline vllm/v1/attention/backends/mla/flashinfer_mla.py]:
```suggestion

```

**Comment by jdebache** [comment]
Closing, was implemented as part of https://github.com/vllm-project/vllm/pull/50000
